bin/lt.js

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env node
+
+const openurl = require('openurl');
+const yargs = require('yargs');
+
+const localtunnel = require('../localtunnel');
+const { version } = require('../package');
+
+const { argv } = yargs
+  .usage('Usage: lt --port [num] <options>')
+  .env(true)
+  .option('h', {
+    alias: 'host',
+    describe: 'Upstream server providing forwarding',
+    default: 'https://localtunnel.me',
+  })
+  .option('s', {
+    alias: 'subdomain',
+    describe: 'Request this subdomain',
+  })
+  .option('l', {
+    alias: 'local-host',
+    describe: 'Tunnel traffic to this host instead of localhost, override Host header to this host',
+  })
+  .options('o', {
+    alias: 'open',
+    describe: 'Opens url in your browser',
+  })
+  .option('p', {
+    alias: 'port',
+    describe: 'Internal http server port',
+  })
+  .option('print-requests', {
+    describe: 'Print basic request info',
+  })
+  .require('port')
+  .boolean('print-requests')
+  .help('help', 'Show this help and exit')
+  .version(version);
+
+if (typeof argv.port !== 'number') {
+  yargs.showHelp();
+  // eslint-disable-next-line no-console
+  console.error('port must be a number');
+  process.exit(1);
+}
+
+(async () => {
+  const tunnel = await localtunnel({
+    host: argv.host,
+    port: argv.port,
+    local_host: argv['local-host'],
+    subdomain: argv.subdomain,
+  }).catch(err => {
+    throw err;
+  });
+
+  tunnel.on('error', err => {
+    throw err;
+  });
+
+  // eslint-disable-next-line no-console
+  console.log('your url is: %s, your cachedUrl is %s', tunnel.url, tunnel.cachedUrl);
```

Comment:
What is a `cachedUrl`? Why should the user care about that versus a regular url? The CLI is suppose to give you just one URL.

Reply:
`cachedUrl` is something returned by our fork of the localtunnel server (which we run at tunnel.chromaticqa.com`. The reason is a bit complicated but here's the gist:

In order to build a horizontally scalable version of the tunnel server using GCP's GKE (Kubernetes) routing system, we needed to setup a situation where the URL you use while the tunnel is open uses a fixed domain. Instead we use a port number for routing (e.g. `tunnel.chromaticqa.com:31485` rather than `xyz.tunnel.chromaticqa.com`). This is fine as long as it's opaque to the user, but we also have to expose a human-friendly URL with a subdomain, so we actually do both.

This second URL is called `cachedUrl` because we needed a way to keep the tunnel URL available even after it has closed (because we have a webapp that exposes it). To do that we upload the tunneled resources to AWS S3 and serve them forever on `cachedUrl`. This is specific to our localtunnel server fork, but I imagine this could be an opt-in (perhaps paid) feature for `localtunnel.me` as well.

In essense you can think of it like so:
- `url` is the actual URL to the tunnel, only available while the tunnel is open. So this is the same behavior as `localtunnel.me`, except that our server fork uses a different URL scheme.
- `cachedUrl` is a publicly available URL to a cached version of the tunneled resources, which remains available after the tunnel closes. Obviously only resources that were sent through the tunnel while it was open are actually cached.

This feature is critically important to the way our systems work. The only thing the client does is pass this value along to the caller, so it's not really impactful. I'll change the logging so it won't log if there is no `cachedUrl`.